WeakAuras/Conditions.lua

```diff
@@ -271,6 +271,52 @@ local function CreateTestForCondition(uid, input, allConditionsTemplate, usedSta
       else
         check = stateCheck .. stateVariableCheck .. "state[" .. trigger .. "]".. string.format("[%q]", variable) .. op .. "'" .. value .. "'";
       end
+    elseif (cType == "range" and value and op and input.type and input.op_range and input.range) then
+      local fn
+      if input.type == "group" then
+        fn = [[
+          return function()
+            local found = 0
+            local op = %q
+            local range = %s
+            for unit in WA_IterateGroupMembers() do
```

Comment:
The reason I took so long was that I'm not entirely happy with this. Each condition iterates all members or enemies to check its status. If there are multiple auras using the same range check, then all of them will use they separate check.

I think there are improvements possible, but they do require a bigger rework of our internal events.

Reply:
Ideally i think it belongs to LibRangeCheck scope.
It should have functions to handle group & nameplate checks, with a cache clean each new frame